### PR TITLE
BAVL-501 adding a bit more logic around what days reminder emails are sent and how far in the future to look.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Reminder emails sent Monday to Thurs and Sunday look a day ahead.
Reminder emails sent Friday look at the following Monday.
Reminder emails are not sent on Saturday.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt [hunk 1]
```diff
@@ -91,7 +91,7 @@ class BookingFacade(
     sendTelemetry(BookingAction.CANCEL, booking, cancelledBy)
   }
 
-  fun courtHearingLinkReminder(videoBooking: VideoBooking, user: User) {
+  fun courtHearingLinkReminder(videoBooking: VideoBooking, user: ServiceUser) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/BookingFacade.kt:94]: Can get away with ServiceUser hear as should not be called by other user types.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/TimeSourceConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.time.Clock
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Configuration
+class TimeSourceConfiguration {
+  @Bean
+  fun timeSource() = TimeSource { LocalDateTime.now(Clock.systemDefaultZone()) }
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/config/TimeSourceConfiguration.kt:12]: This gives me better control of time when testing.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/jobs/CourtHearingLinkReminderJobTest.kt [hunk 3]
```diff
@@ -6,64 +6,108 @@ import org.mockito.kotlin.doReturn
...
-
-    courtHearingLinkReminderJob.block()
-
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(TUESDAY)) doReturn booking.appointments()
+    runJobOn(MONDAY)
     verify(bookingFacade, never()).courtHearingLinkReminder(any(), any())
   }
+
+  private fun runJobOn(date: LocalDate) {
+    CourtHearingLinkReminderJob(prisonAppointmentRepository, bookingFacade) { date.atStartOfDay() }.runJob()
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/jobs/CourtHearingLinkReminderJobTest.kt:111]: This is setting the date on the TimeSource passed into the job making it possible to test each day.

**Review by timharrison-moj — DISMISSED** [review]
LGTM

File: helm_deploy/values-dev.yaml [hunk 4]
```diff
@@ -32,4 +32,4 @@ generic-prometheus-alerts:
     - "hmpps-book-a-video-link-dev-hmpps_book_a_video_link_domain_dlq"
 
 cron:
-  courtHearingLinkReminderJob: "0 15 * * 1-5"
+  courtHearingLinkReminderJob: "0 11 * * 1-5"
```
Comment (matt-crowson-moj) [inline helm_deploy/values-dev.yaml:35]: Brought forwards in DEV to aid testing prior to prod run.

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**